packages/sync/src/modules/Module.php

```diff
@@ -373,4 +396,64 @@ public function get_objects_by_id( $object_type, $ids ) {
 
 		return $objects;
 	}
+
+	/**
+	 * Gets a list of minimum and maximum object ids for each batch based on the given batch size.
+	 *
+	 * @access public
+	 *
+	 * @param int         $batch_size The batch size for objects.
+	 * @param string|bool $where_sql  The sql where clause minus 'WHERE', or false if no where clause is needed.
+	 * @param bool        $distinct   True if we should only look at distinct object ids.
+	 *
+	 * @return array|bool An array of min and max ids for each batch.
+	 */
+	public function get_min_max_object_ids_for_batches( $batch_size, $where_sql = false, $distinct = false ) {
+		global $wpdb;
+
+		if ( ! $this->table_name() ) {
+			return false;
+		}
+
+		$results      = array();
+		$table        = $wpdb->{$this->table_name()};
+		$current_max  = 0;
+		$current_min  = 1;
+		$id_field     = $this->id_field();
+		$replicastore = new Replicastore();
+
+		$total = $replicastore->get_min_max_object_id(
+			$id_field,
+			false,
+			$table,
+			$where_sql,
+			false
+		);
+
+		while ( $total->max > $current_max ) {
+			$where  = $where_sql ?
+				$where_sql . "AND $id_field > $current_max" :
+				"$id_field > $current_max";
+			$result = $replicastore->get_min_max_object_id(
+				$id_field,
+				$distinct,
+				$table,
+				$where,
+				$batch_size
+			);
+			if ( empty( $result->min ) && empty( $result->max ) ) {
+				$current_max = (int) $total->max;
```

Comment:
This line could use a comment; took me some time to understand what we're doing here.